doc/src/extras.rst

```diff
@@ -141,6 +141,262 @@ Logging cursor
 
 .. autoclass:: MinTimeLoggingCursor
 
+Replication cursor
+^^^^^^^^^^^^^^^^^^
+
+.. autoclass:: ReplicationConnection
+
+   This connection factory class can be used to open a special type of
+   connection that is used for streaming replication.
+
+   Example::
+
+       from psycopg2.extras import ReplicationConnection, REPLICATION_PHYSICAL, REPLICATION_LOGICAL
+       conn = psycopg2.connect(dsn, connection_factory=ReplicationConnection)
+       cur = conn.cursor()
+
+.. seealso::
+
+   - PostgreSQL `Replication protocol`__
+
+   .. __: http://www.postgresql.org/docs/current/static/protocol-replication.html
+
+.. autoclass:: ReplicationCursor
+
+    .. method:: identify_system()
+
+       Execute ``IDENTIFY_SYSTEM`` command of the streaming replication
+       protocol and return the result as a dictionary.
+
+       Example::
+
+           >>> cur.identify_system()
+           {'timeline': 1, 'systemid': '1234567890123456789', 'dbname': 'test', 'xlogpos': '0/1ABCDEF'}
+
+    .. method:: create_replication_slot(slot_type, slot_name, output_plugin=None)
+
+       Create streaming replication slot.
+
+       :param slot_type: type of replication: either `REPLICATION_PHYSICAL` or
+                         `REPLICATION_LOGICAL`
+       :param slot_name: name of the replication slot to be created
+       :param output_plugin: name of the logical decoding output plugin to use
+                             (logical replication only)
+
+       Example::
+
+           cur.create_replication_slot(REPLICATION_LOGICAL, "testslot", "test_decoding")
+
+    .. method:: drop_replication_slot(slot_name)
+
+       Drop streaming replication slot.
+
+       :param slot_name: name of the replication slot to drop
+
+       Example::
+
+           cur.drop_replication_slot("testslot")
+
+    .. method:: start_replication(slot_type, slot_name=None, writer=None, start_lsn=None, timeline=0, keepalive_interval=10, options=None)
+
+       Start a replication stream.  On non-asynchronous connection, also
+       consume the stream messages.
+
+       :param slot_type: type of replication: either `REPLICATION_PHYSICAL` or
+                         `REPLICATION_LOGICAL`
+       :param slot_name: name of the replication slot to use (required for
+                         logical replication)
+       :param writer: a file-like object to write replication messages to
+       :param start_lsn: the LSN position to start from, in the form
+                         ``XXX/XXX`` (forward-slash separated pair of
+                         hexadecimals)
```

Comment:
That LSN is really just a uint64 formatted crazily for historical reasons. So an `int` should be acceptable too. See the implementation of the `pg_lsn` data type on the server side in 9.4+ for details and example code.

Needs extra text like:

> The LSN `0` (`'0/0'` in text format) may be passed to request that replay begins at the earliest available point.

and maybe:

> Note that replay may begin at a point later in history than the passed LSN if the data for that LSN is no longer present on the server. The client must ensure that repay has begun at the expected position using information in the COPY BOTH protocol stream.

... or just a warning to read the docs in detail there.